### PR TITLE
HDDS-10058. Replace Hamcrest assertions

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Replace Hamcrest `assertThat` and matchers with JUnit or AssertJ assertions, to reduce the number of test dependencies a bit.

https://issues.apache.org/jira/browse/HDDS-10058

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/7409626572

### CODE REVIEW

**Review by hemantk-12 — APPROVED** [review]
Thanks for the patch @adoroszlai .

LGTM+1.

File: pom.xml [hunk 1]
```diff
@@ -1282,7 +1282,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
       </dependency>
       <dependency>
         <groupId>org.hamcrest</groupId>
-        <artifactId>hamcrest-all</artifactId>
+        <artifactId>hamcrest-core</artifactId>
```
Comment (kerneltime) [inline pom.xml:1285]: Can we try to get rid of core as well?

Comment (adoroszlai) [reply pom.xml:1285]: Sure, let me try that.

Comment (kerneltime) [reply pom.xml:1285]: I see https://github.com/apache/ozone/pull/5923/

Comment (adoroszlai) [reply pom.xml:1285]: Yes, we can remove `hamcrest-core`, but only because we also have `hamcrest` as a transitive dependency.

**Pull request merged**